File: php/27-remove-element.php
<?php
class Solution {
    /**
     * @param Integer[] $nums
     * @param Integer $val
     * @return Integer
     */
    function removeElement(&$nums, $val) {
        $count = count($nums);
        $mewLen = $count;
        for ($i = 0; $i < $count; $i++) {
            if ($nums[$i] == $val) {
                unset($nums[$i]);
                $newLen--;
            }
        }
        return $newLen;
    }
}

File: go/27-remove-element.go
package main

func removeElement(nums []int, val int) int {
	count := -1
	for i := 0; i < len(nums); i++ {
		if nums[i] != val {
			count++
			nums[count] = nums[i]
		}
	}
	return count + 1
}
